### ISSUE TITLE
click shift on 1080x2400 display

### ISSUE THREAD

**Issue by tkkcc** [rib/android-activity#43]
agtk-egui

https://user-images.githubusercontent.com/17373509/203543713-e6893c36-cb8b-4aab-9ae3-f2db812a8968.mp4

**Comment by rib** [comment]
Oh, odd, I don't think I've seen that when I run the same example - but maybe I was testing `agdk-eframe` more recently.

Thanks for the report.

This is maybe something that should ideally be tracked as either a Winit or an Egui issue but it's probably fine here for now until we determine what the underlying issue really is - considering I've been working on Winit and Android support in Egui too.

I'm not 100% sure when I'll get around to investigating so please post here if you figure out any more clues from experimenting yourself.

It could also be helpful to know whether you see the same thing with `agdk-eframe`

I wonder if it relates to how DPI scaling is done, and maybe the input is being scaled by the wrong factor (not consistent with how the visuals are scaled by DPI)

At some point I think the example may have had some hacky 2x multiplier for the DPI because the regular DPI scaling was still resulting in an unusably small UI on mobile phones - maybe I left a constant scale factor for rendering in the example that's not also used for input events.

**Comment by tkkcc** [comment]
It's tested on a poco f3 phone with official system. no mod no root. DPI is 440:
![Screenshot_2022-11-27-22-51-43-409_com liuzh deviceinfo](https://user-images.githubusercontent.com/17373509/204141845-f0d7fb25-6c3f-490e-9aa3-761f0015c29b.jpg)

agdk-eframe has same problem. egui.rs in webbrowser works fine. 

I also tried on genymotion emulators but app stops on boot, logcat shows error about egl. The arch is x86, I also gaming on it.

**Comment by tkkcc** [comment]
I can reproduce this issue on waydroid on another PC, `agdk-eframe` with ndk 25.1.8937393 or 24.0.8215888 .

**Comment by MarijnS95** [comment]
Since there doesn't seem to be an issue horizontally, and the window _appears to_ move at the same speed as the cursor (hard to tell from a minimal video), is this perhaps simply the offset of the titlebar not being taken into account?

**Comment by rib** [comment]
Sounds very plausible yep. I had started looking at updating the winit backend to add support for querying the surface size distinct from the inner size that would be able to account for insets (re: https://github.com/rust-windowing/winit/issues/2308) and maybe that'll end up helping enable egui to consider insets for a titlebar.

**Comment by MarijnS95** [comment]
Yeah it looks very much that this isn't the surface size but the whole window/`Activity` size. Would be great to have a repro project to validate this theory (again: not visible from a video...), though can probably whip one up easily.

**Comment by tkkcc** [comment]
add `getSupportActionBar().hide();` after `hideSystemUI();` can solve this. just remove the native bar.
